src/core/network/qgsnetworkaccessmanager.cpp

```diff
@@ -434,16 +427,17 @@ void QgsNetworkAccessManager::onReplySslErrors( const QList<QSslError> &errors )
 
   emit requestEncounteredSslErrors( getRequestId( reply ), errors );
 
+  mSslErrorHandlerSemaphore.acquire();
+
   // in main thread this will trigger SSL error handler immediately and return once the errors are handled,
   // while in worker thread the signal will be queued (and return immediately) -- hence the need to lock the thread in the next block
   emit sslErrorsOccurred( reply, errors );
   if ( this != sMainNAM )
   {
     // lock thread and wait till error is handled. If we return from this slot now, then the reply will resume
     // without actually giving the main thread the chance to act on the ssl error and possibly ignore it.
-    mSslErrorHandlerMutex.lock();
-    mSslErrorWaitCondition.wait( &mSslErrorHandlerMutex );
-    mSslErrorHandlerMutex.unlock();
+    mSslErrorHandlerSemaphore.acquire();
```

Comment:
It's already acquired outside the conditional block if I read correctly

Reply:
Yes, that's the point! You can read more in this [PR](https://github.com/qgis/QGIS/pull/45833), that's kind of the same issue

The goal is to block until the semaphore is released in *afterAuthRequestHandled*

Reply:
Thanks, it's really not intuitive to understand two lines being .acquire() and .release(), maybe add detailed comments to explain the rationale for that ?